src/bridge/test-websocketstream.c

```diff
@@ -169,7 +169,6 @@ test_basic (TestCase *test,
 
   recv = mock_transport_pop_channel (test->transport, "444");
   cockpit_assert_bytes_eq (recv, "MESSAGE", 7);
-  g_bytes_unref (recv);
```

Comment:
Would it be possible to let the compiler find that? By changing `mock_transport_pop_channel` to return a `const GBytes*`?

Reply:
For better or worse, this isn't how things are done in GLib world.  Even the "consty" accessors to `GBytes` (like `g_bytes_get_size()`) take a `GBytes *` (instead of a `const GBytes *`) which means that we'd pretty much not be able to do anything at all with the return value.

This is why we have to sprinkle "transfer" annotations all over the place...

It's worth noting that we go well out of our way to accommodate this being `(transfer none)`.  We maintain a trash stack inside of the mock transport for this purpose, to make sure that the object is not transfered, but stays around even after it's popped (for inspection).  That pretty much always saves us some typing in test cases...

Reply:
Thanks for the explanation!